crates/extrinsics/src/lib.rs

```diff
@@ -238,7 +238,7 @@ async fn dry_run_extrinsic<C, Call, Signer>(
     rpc: &LegacyRpcMethods<C>,
     call: &Call,
     signer: &Signer,
-) -> core::result::Result<DryRunResult, subxt::Error>
+) -> core::result::Result<DryRunResultBytes, subxt::Error>
```

Comment:
New type wraps the bytes, allowing lazy conversion into a `DryRunResult` with new `DryRunDecodeError` which is used elsewhere.